bgpd/bgp_attr.c

```diff
@@ -3016,39 +3005,23 @@ bgp_attr_psid_sub(uint8_t type, uint16_t length,
 		attr->srv6_vpn->sid_flags = sid_flags;
 		sid_copy(&attr->srv6_vpn->sid, &ipv6_sid);
 		attr->srv6_vpn = srv6_vpn_intern(attr->srv6_vpn);
-	}
-
-	/* Placeholder code for the SRv6 L3 Service type */
-	else if (type == BGP_PREFIX_SID_SRV6_L3_SERVICE) {
-		if (STREAM_READABLE(peer->curr) < length) {
+	} else if (type == BGP_PREFIX_SID_SRV6_L3_SERVICE) {
+		if (STREAM_READABLE(peer->curr) != 1) {
```

Comment:
Shouldn't be `<1` here?

We are processing the `BGP_PREFIX_SID_SRV6_L3_SERVICE` TLV, which has a 1-byte RESERVED field and a variable-length SRv6 Service Sub-TLVs field.

Here we ensure that we have at least 1 byte (for the RESERVED field). And then we call `bgp_attr_srv6_service()` to validate and parse the SRv6 Service Sub-TLVs.

Reply:
I'm wondering if this is related to the failing topo tests, as well ... it looks like routes are missing, which makes me think we're not processing all the routes being sent to us (?)

Reply:
yep fixed